components/keys/src/types.rs

```diff
@@ -88,6 +88,13 @@ impl Key {
         Key(encoded)
     }
 
+    #[inline]
+    pub fn append_ts_ref(&mut self, ts: u64) -> &Key {
```

Comment:
Honestly I don't like the newly introduced interfaces. They violate the current design, which is key is immutable. A good way is to completely abandon these immutable interfaces, instead of making them a mixture of immutable and mutable.

Reply:
Cry... How about using `mem::replace`?